main.go

```diff
@@ -318,20 +318,73 @@ func printConfigProblems() {
 // system running the verifier is greater than or equal to 8.1;
 // if not then prints a warning
 func checkOpenSSHVersion() {
-	// Redhat/centos does not recognize `sshd -V` but does recognize `ssh -V`
-	// Ubuntu recognizes both
-	cmd := exec.Command("ssh", "-V")
-	output, err := cmd.CombinedOutput()
-	if err != nil {
-		log.Println("Warning: Error executing ssh -V:", err)
+	version := getOpenSSHVersion()
+	if version == "" {
+		log.Println("Warning: Could not determine OpenSSH version")
 		return
 	}
 
-	if ok, _ := isOpenSSHVersion8Dot1OrGreater(string(output)); !ok {
+	if ok, _ := isOpenSSHVersion8Dot1OrGreater(version); !ok {
 		log.Println("Warning: OpenPubkey SSH requires OpenSSH v. 8.1 or greater")
 	}
 }
 
+// getOpenSSHVersion attempts to get OpenSSH version using multiple fallback methods
+func getOpenSSHVersion() string {
+	// Try ssh -V first (works on most systems)
+	cmd := exec.Command("ssh", "-V")
+	output, err := cmd.CombinedOutput()
+	if err == nil {
+		return string(output)
+	}
+	log.Println("Warning: Error executing ssh -V:", err)
+
+	// Try sshd -V as fallback
+	cmd = exec.Command("sshd", "-V")
+	output, err = cmd.CombinedOutput()
+	if err == nil {
+		return string(output)
+	}
+	log.Println("Warning: Error executing sshd -V:", err)
+
+	// Fall back to OS-specific package manager queries
+	osType := detectOS()
+	log.Printf("Attempting OS-specific version detection for: %s", osType)
+
+	switch osType {
+	case OSTypeRHEL:
+		// For RedHat-based systems (CentOS, RHEL, Fedora)
+		cmd = exec.Command("sh", "-c", "version=$(rpm -q --qf \"%{VERSION}\\n\" openssh-server | sed -E 's/^([0-9]+\\.[0-9]+).*/\\1/'); echo \"OpenSSH_$version\"")
```

Comment:
Is there any chance someone could make a script called `rpm` which is earlier in the PATH than the actual rpm and then use this to run arbitrary shell instructions at the opkssh user?

I think a full path to each of the commands `/usr/bin/rpm` should harden against this attack.